Licence: same as the licence of the upstream package.

diff -ur acpid.8 acpid.8
--- acpid.8	2014-08-22 21:12:22.029473237 +0800
+++ acpid.8	2018-02-25 15:29:33.721911979 +0800
@@ -82,9 +82,7 @@
 can be made to the \fBacpid\fP socket.  Default is \fI256\fP.
 .TP 12
 .BI \-d "\fR, \fP" \-\-debug
-This option increases the \fBacpid\fP debug level by one.  If the debug level
-is non-zero, \fBacpid\fP will run in the foreground, and will log to
-stderr, in addition to the regular syslog.
+This option increases the \fBacpid\fP debug level by one.
 .TP
 .BI \-e "\fR, \fP" \-\-eventfile " filename"
 This option changes the event file from which \fBacpid\fP reads events.
@@ -94,7 +92,8 @@
 This option forces \fBacpid\fP to use the Linux kernel input layer and netlink interface for ACPI events.
 .TP
 .BI \-f "\fR, \fP" \-\-foreground
-This option keeps \fBacpid\fP in the foreground by not forking at startup.
+This option keeps \fBacpid\fP in the foreground by not forking at startup,
+and makes it log to stderr instead of syslog.
 .TP
 .BI \-l "\fR, \fP" \-\-logevents
 This option tells \fBacpid\fP to log information about all events and actions.
diff -ur acpid.c acpid.c
--- acpid.c	2016-04-25 07:13:22.996715770 +0800
+++ acpid.c	2018-02-25 15:45:55.823976358 +0800
@@ -52,9 +52,6 @@
 static void clean_exit(int sig);
 static void reload_conf(int sig);
 
-/* global debug level */
-int acpid_debug;
-
 /* do we log event info? */
 int logevents;
 
@@ -82,7 +79,10 @@
 	close_fds();
 
 	/* open the log */
-	open_log();
+	if (foreground)
+		log_to_stderr = 1;
+	else
+		open_log();
 
 	/* if we're running in the background, and we're not being started */
 	/* by systemd */
@@ -218,7 +218,7 @@
 	const char *opts_help[] = {
 		"Set the configuration directory.",	/* confdir */
 		"Set the limit on non-root socket connections.",/* clientmax */
-		"Increase debugging level (implies -f).",/* debug */
+		"Increase debugging level.",/* debug */
 		"Use the specified file for events.",	/* eventfile */
 		"Run in the foreground.",		/* foreground */
 		"Log all event activity.",		/* logevents */
@@ -253,9 +253,7 @@
 			clientmax = strtol(optarg, NULL, 0);
 			break;
 		case 'd':
-			foreground = 1;
 			acpid_debug++;
-            log_debug_to_stderr = 1;
 			break;
 		case 'e':
 			eventfile = optarg;
@@ -392,9 +390,6 @@
 
 	/* open the syslog */
 	log_opts = LOG_CONS|LOG_NDELAY;
-	if (acpid_debug) {
-		log_opts |= LOG_PERROR;
-	}
 	openlog(PACKAGE, log_opts, LOG_DAEMON);
 }
 
@@ -419,12 +414,12 @@
 		close(nullfd);
 		return -1;
 	}
-	if (!acpid_debug && dup2(nullfd, STDOUT_FILENO) != STDOUT_FILENO) {
+	if (!log_to_stderr && dup2(nullfd, STDOUT_FILENO) != STDOUT_FILENO) {
 		acpid_log(LOG_ERR, "dup2() stdout: %s", strerror(errno));
 		close(nullfd);
 		return -1;
 	}
-	if (!acpid_debug && dup2(nullfd, STDERR_FILENO) != STDERR_FILENO) {
+	if (!log_to_stderr && dup2(nullfd, STDERR_FILENO) != STDERR_FILENO) {
 		acpid_log(LOG_ERR, "dup2() stderr: %s", strerror(errno));
 		close(nullfd);
 		return -1;
diff -ur acpid.h acpid.h
--- acpid.h	2014-02-26 08:36:58.788146100 +0800
+++ acpid.h	2018-02-25 15:52:54.085170824 +0800
@@ -42,7 +42,6 @@
 /*
  * acpid.c
  */
-extern int acpid_debug;
 extern int logevents;
 extern const char *progname;
 extern int tpmutefix;
diff -ur log.c log.c
--- log.c	2011-10-18 06:52:32.420965575 +0800
+++ log.c	2018-02-25 15:47:05.438342974 +0800
@@ -26,7 +26,8 @@
 
 #include "log.h"
 
-int log_debug_to_stderr = 0;
+int log_to_stderr = 0;
+int acpid_debug = 0;
 
 int
 #ifdef __GNUC__
@@ -34,23 +35,17 @@
 #endif
 acpid_log(int level, const char *fmt, ...)
 {
+	if (level == LOG_DEBUG && !acpid_debug) return 0;
 	va_list args;
+	va_start(args, fmt);
 
-	if (level == LOG_DEBUG) {
-		/* if "-d" has been specified */
-		if (log_debug_to_stderr) {
-			/* send debug output to stderr */
-			va_start(args, fmt);
-			vfprintf(stderr, fmt, args);
-			va_end(args);
-
-            fprintf(stderr, "\n");
-		}
+	if (log_to_stderr) {
+		vfprintf(stderr, fmt, args);
+		fprintf(stderr, "\n");
 	} else {
-		va_start(args, fmt);
 		vsyslog(level, fmt, args);
-		va_end(args);
 	}
 
+	va_end(args);
 	return 0;
 }
diff -ur log.h log.h
--- log.h	2012-02-21 09:26:37.514721963 +0800
+++ log.h	2018-02-25 15:48:24.322625249 +0800
@@ -26,11 +26,11 @@
 /* for LOG_ERR, LOG_DEBUG, LOG_INFO, etc... */
 #include <syslog.h>
 
-/*
- * Set to 1 to send LOG_DEBUG logging to stderr, zero to ignore LOG_DEBUG
- * logging.  Default is zero.
- */
-extern int log_debug_to_stderr;
+/* Set to 1 to send logging to stderr.  Default is zero. */
+extern int log_to_stderr;
+
+/* Global debug level. */
+extern int acpid_debug;
 
 extern int acpid_log(int level, const char *fmt, ...) __attribute__((format(printf,2,3)));
 
